lib/teleterm/apiserver/handler/handler_access_requests.go

```diff
@@ -0,0 +1,155 @@
+	requests, err := s.DaemonService.GetAccessRequests(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	response := &api.GetAccessRequestsResponse{}
+	for _, req := range requests {
+		response.Requests = append(response.Requests, newApiAccessRequest(req))
+	}
+
+	return response, nil
+}
+
+// GetAccessRequest returns a single access request by id
+func (s *Handler) GetAccessRequest(ctx context.Context, req *api.GetAccessRequestsRequest) (*api.GetAccessRequestResponse, error) {
+	requests, err := s.DaemonService.GetAccessRequests(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	response := &api.GetAccessRequestResponse{}
+	if len(requests) < 1 {
+		return nil, trace.NotFound("access request %q not found", req.Id)
+	}
+	response.Request = newApiAccessRequest(requests[0])
+
+	return response, nil
+}
+
+func (s *Handler) CreateAccessRequest(ctx context.Context, req *api.CreateAccessRequestRequest) (*api.CreateAccessRequestResponse, error) {
+	request, err := s.DaemonService.CreateAccessRequest(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	createdRequest := &api.CreateAccessRequestResponse{
+		Request: newApiAccessRequest(*request),
+	}
+	return createdRequest, nil
+}
+
+func (s *Handler) DeleteAccessRequest(ctx context.Context, req *api.DeleteAccessRequestRequest) (*api.DeleteAccessRequestResponse, error) {
+	err := s.DaemonService.DeleteAccessRequest(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &api.DeleteAccessRequestResponse{
+		Success: "true",
+	}, nil
+}
+
+func (s *Handler) AssumeRole(ctx context.Context, req *api.AssumeRoleRequest) (*api.AssumeRoleResponse, error) {
+	err := s.DaemonService.AssumeRole(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &api.AssumeRoleResponse{
+		Success: "true",
+	}, nil
+
+}
+
+func (s *Handler) ReviewAccessRequest(ctx context.Context, req *api.ReviewAccessRequestRequest) (*api.ReviewAccessRequestResponse, error) {
+	request, err := s.DaemonService.ReviewAccessRequest(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	response := &api.ReviewAccessRequestResponse{
+		Request: newApiAccessRequest(*request),
+	}
+	return response, nil
+
+}
+
+func newApiAccessRequest(req clusters.AccessRequest) *api.AccessRequest {
+	reviews := []*api.AccessRequestReview{}
+	requestReviews := req.GetReviews()
+	for _, rev := range requestReviews {
+
+		reviews = append(reviews, &api.AccessRequestReview{
+			Author:  rev.Author,
+			Roles:   rev.Roles,
+			State:   rev.ProposedState.String(),
+			Reason:  rev.Reason,
+			Created: rev.Created.String(),
+		})
+	}
+
+	thresholdNames := make([]string, 0, len(req.GetThresholds()))
+	for _, t := range req.GetThresholds() {
+		if t.Name != "" {
+			thresholdNames = append(thresholdNames, t.Name)
+		}
+	}
+
+	requestedResourceIDs := make([]*api.ResourceID, 0, len(req.GetRequestedResourceIDs()))
+	for _, r := range req.GetRequestedResourceIDs() {
```

Comment:
I recently added some logic for the Web UI to get extra details about requested resources, currently just SSH server hostnames and in the future we'll want resource labels as well https://github.com/gravitational/teleport.e/pull/530/files

I don't think you should add it to this PR but it will probably be desirable to have in Connect as well. Would be nice if we could share code between the web and connect backends for this

Reply:
@avatus could you file an issue to add support for this?

Reply:
https://github.com/gravitational/webapps.e/issues/406